django/db/models/query.py

```diff
@@ -981,7 +982,10 @@ def annotate(self, *args, **kwargs):
         clone = self._chain()
         names = self._fields
         if names is None:
-            names = {f.name for f in self.model._meta.get_fields()}
+            names = set(chain.from_iterable(
+                (field.name, field.attname) if hasattr(field, 'attname') else (field.name,)
+                for field in self.model._meta.get_fields()
+            ))
```

Comment:
This logic above is based on the old code of `get_all_field_names`, see: https://docs.djangoproject.com/en/2.0/ref/models/meta/